pkg/webhook/spc.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"fmt"
+	"net/http"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/pkg/errors"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+func (wh *webhook) validateSPCDeleteRequest(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	response := NewAdmissionResponse().
+		SetAllowed().
+		WithResultAsSuccess(http.StatusAccepted).AR
+
+	cspList, err := wh.clientset.OpenebsV1alpha1().CStorPools().List(
+		metav1.ListOptions{
+			LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + req.Name,
+		})
+	if err != nil {
+		err = errors.Wrapf(err, "could not list csp for spc %s", req.Name)
+		klog.Error(err)
+		response = BuildForAPIObject(response).UnSetAllowed().WithResultAsFailure(err, http.StatusBadRequest).AR
+		return response
+	}
+	fmt.Printf("%+v", cspList.Items)
+	for _, cspObj := range cspList.Items {
+		// list cvrs in all namespaces
+		cvrList, err := wh.clientset.OpenebsV1alpha1().CStorVolumeReplicas("").List(metav1.ListOptions{
+			LabelSelector: string(apis.CStorPoolKey) + "=" + cspObj.Name,
+		})
+		if err != nil {
+			err = errors.Wrapf(err, "Could not list cvr for csp %s", cspObj.Name)
+			response = BuildForAPIObject(response).UnSetAllowed().WithResultAsFailure(err, http.StatusBadRequest).AR
+			return response
+		}
+		if len(cvrList.Items) != 0 {
```

Comment:
we have to consider the case of migration here , add the `skip-validations` to skip the SPC  delete request validation during migration

Reply:
Done